release/tar/linux/opensearch-tar-build.sh

```diff
@@ -0,0 +1,175 @@
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# About:        This shell script generate an opensearch tarball with all the provided plugins installed and ready to use. 
+#               A manifest file and right permissions set up is essential to download the required artifacts
+# Dependencies: yq (More info: https://github.com/mikefarah/yq/tree/v4.4.1#install)
+# Usage: .      /opensearch-tar-build.sh -f <path/to/manifest/file/>
+#
+
+set -e
+
+function usage() {
+    echo "Usage: $0 [args]"
+    echo -e "Supported manifest schema-version: 1.0"
+    echo ""
+    echo "Required arguments:"
+    echo -e "-f MANIFEST_FILE\tPath/to/manifest.yml file"
+    echo ""
+    echo "Optional arguments:"
+    echo -e "-p PLATFORM\tSpecify the platform. eg: linux, macos, windows. Currently only supports and defaults to linux"
+    echo -e "-a ARCHITECTURE\tSpecify the architecture, e.g. x64, arm64. Currently only supports x64 and arm64. Defaults to x64"
+    echo -e "-h help"
+}
+
+while getopts ":ha:p:f:" arg; do
+    case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        p)
+            PLATFORM=$(echo "$OPTARG" | tr '[:upper:]' '[:lower:]')
+            ;;
+        a)
+            ARCHITECTURE=$(echo "$OPTARG" | tr '[:upper:]' '[:lower:]')
+            ;;
+        f)
+            MANIFEST_FILE=$OPTARG
+            ;;
+        :)
+            echo "Error: -${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$PLATFORM" ]; then
+    PLATFORM="linux"
+fi
+
+if [ "$PLATFORM" != "linux" ]; then
+    echo "error: $PLATFORM platform is not supported yet!"
+    usage
+    exit 1
+fi
+
+if [ -z "$ARCHITECTURE" ]; then
+    ARCHITECTURE="x64"
+fi
+
+if [ "$ARCHITECTURE" != "x64" ] && [ "$ARCHITECTURE" != "arm64" ]
+then
+    echo "error: $ARCHITECTURE architecture is not supported yet!"
+    usage
+    exit 1
+fi
+
+if [ -z "$MANIFEST_FILE" ]; then
+    echo "error: You must specify the manifest file"
+    usage
+    exit 1
+fi
+
+if [ ! -f $MANIFEST_FILE ]; then
+    echo "error: The given file doesnot exist. Please check the specified path"
+    usage
+    exit 1
+fi
+
+SCHEMA_VERSION=`yq eval '.manifest.schema-version' $MANIFEST_FILE`
+echo $SCHEMA_VERSION
+
+if [ $SCHEMA_VERSION != 1.0 ]; then
+    echo "error: This script only supports manifest schema version 1.0. Please use the manifest file with right schema version"
```

Comment:
Awesome :)

My bash-fu is weak; do you need to put `1.0` in quotes?

Might be worth putting `$SCHEMA_VERSION` in quotes, too, in case it contains a space.

Reply:
Done